src/App.tsx

```diff
@@ -1,61 +1,155 @@
+import { useEffect, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [hasUserError, setHasUserError] = useState(false);
+  const [hasTodoTitleError, setHasTodoTitleError] = useState(false);
+
+  const getTodosWithUsers = () => {
+    const todosWithUsersToSet = [...todosFromServer]
+      .map(todo => {
+        const user = usersFromServer
+          .filter(({ id }) => id === todo.userId)[0];
+        const { name, id } = user;
+
+        return {
+          ...todo,
+          name,
+          userId: id,
+          user,
+        };
+      });
+
+    return todosWithUsersToSet;
+  };
+
+  useEffect(() => {
+    const todosToSet = getTodosWithUsers();
+
+    setTodos(todosToSet);
+  }, []);
+
+  const getMaxTodoId = () => {
+    return todos
+      .sort((todoOne, todoTwo) => todoTwo.id - todoOne.id)[0].id || -1;
```

Comment:
let's use reduce or Math.max + map to reduce complexity

Reply:
done with Math.max(todos.map(({id}) => id))